src/reducers/index.js

```diff
@@ -372,14 +372,13 @@ const onUpdateLoadedData = (state, action) => ({
 });
 
 const onCleanData = (state) => {
-  if (state.wallet) {
-    state.wallet.stop();
-  }
-
   return Object.assign({}, initialState, {
     isVersionAllowed: state.isVersionAllowed,
     loadingAddresses: state.loadingAddresses,
     ledgerWasClosed: state.ledgerWasClosed,
+    // Keep the unleashClient as it should continue running
+    unleashClient: state.unleashClient,
+    featureTogglesInitialized: state.featureTogglesInitialized,
```

Comment:
If not persisted, the `unleashClient` and `featureTogglesInitialized` may lock the wallet in an infinite wait period on the next start, since we wait for an event to set the featureTogglesInitialized but it will never come.

Reply:
I agree, the reason for this is that some variables in store are loaded by sagas that run only on startup and would not get reloaded after a wallet reset and in this case, adding a passphrase.

I'll create a ktlo for this.